src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileStatusService.scala

```diff
@@ -0,0 +1,26 @@
+package uk.gov.nationalarchives.tdr.api.service
+
+import uk.gov.nationalarchives.tdr.api.db.repository.FileStatusRepository
+import uk.gov.nationalarchives.tdr.api.service.FileStatusService.Success
+
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future}
+
+class FileStatusService(fileStatusRepository: FileStatusRepository)(implicit val executionContext: ExecutionContext) {
+
+  def allChecksSucceeded(consignmentId: UUID): Future[Boolean] = {
+    for {
+      checksumStatus <- fileStatusRepository.getFileStatus(consignmentId, "checksum")
+    } yield checksumStatus.headOption.exists(_.value == Success)
+  }
+
+}
+
+object FileStatusService {
+  //Status types
+  val Checksum = "checksum"
```

Comment:
This should be capitalized: `Checksum`. That's my fault, sorry - I copied it manually to the Jira ticket.

Reply:
It's ok. I've changed it.